framework/include/vectorpostprocessors/MeshDivisionFunctorReductionVectorPostprocessor.h

```diff
@@ -0,0 +1,58 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "ElementVectorPostprocessor.h"
+#include "NonADFunctorInterface.h"
+
+/**
+ * This MeshDivisionFunctorReductionVectorPostprocessor serves to integrate functors based
+ * on the index of the elements in the mesh division
+ */
+class MeshDivisionFunctorReductionVectorPostprocessor : public ElementVectorPostprocessor,
```

Comment:
Aren't we supposed to use reporters instead of vector postprocessors?

Reply:
I dont know? I would think VPPs are fine since:

We can get vectors from VPP using the reporter interface or the VPPInterface
and VPPs are a good concept user-wise

Reply:
@loganharbour you are a codeowner for reporters. What say you?